src/main/java/Duke.java

```diff
@@ -1,10 +1,56 @@
+import duke.backend.Storage;
+import duke.frontend.Parser;
+import duke.frontend.Ui;
+import duke.task.TaskList;
+
+import java.io.FileNotFoundException;
+
+/**
+ * Runs the main method that in turn instantiates and runs the Duke object.
+ *
+ * The program is a CLI Application that allows users to:
+ * <ul>
+ *     <li>Add new tasks (Deadline, Event or Todo)</li>
+ *     <li>View a list of their tasks</li>
+ *     <li>Mark tasks as done</li>
+ *     <li>Delete tasks</li>
+ *     <li>Save tasks in the hard disk automatically</li>
+ * </ul>
+ *
+ * @author Damith C. Rajapakse
+ * @author Jeffry Lum
+ * @author Jiachen
+ * @author Jeanne Toh
+ */
 public class Duke {
+    private Storage storage;
+
+    private TaskList taskList;
+    private Parser parser;
+
+    private Duke(String filePath) {
+        try {
+            storage = new Storage(filePath);
+            taskList = new TaskList(storage.load());
+            parser = new Parser(taskList);
+        } catch (FileNotFoundException e) {
+            System.out.println("Error: file not created. " + e);
+        }
+    }
+
+    private void run() {
+        Ui.greet();
+        this.parser.parseInputCommands();
+        this.storage.save(this.taskList);
```

Comment:
Should you remove the `this` keyword? The [java coding standard documentation](https://se-education.org/guides/conventions/java/index.html#variables) mentions to use `this` keyword only when a field is shadowed by a method or constructor parameter.

Reply:
Edited accordingly. Thank you! :) Appreciate this comment.